### PR TITLE
FIX Folder names no longer allow dots, and are replaced with dashes

### PR DESCRIPTION
**Pull request by robbieaverill**
Fixes https://github.com/silverstripe/silverstripe-asset-admin/issues/830

### CODE REVIEW

File: src/Folder.php [hunk 1]
```diff
@@ -109,6 +109,25 @@ public function getIcon()
...
+    /**
+     * Use default File filter rules, but also remove dots and replace with dashes.
+     *
+     * @param string $name
+     * @return string
+     */
+    protected function filterFilename($name)
+    {
+        // Fix illegal characters
+        $filter = FileNameFilter::create();
```
Comment (tractorcow) [inline src/Folder.php]: Would you be better off with a FileNameFilter subclass (e.g. named service) custom to folders? This is un-injectable.

Comment (robbieaverill) [reply src/Folder.php]: True, fair enough

Comment (robbieaverill) [reply src/Folder.php]: I've done this now =)

**Comment by robbieaverill** [comment]
We really need a FilterInterface for these classes, so I've added one for future consumption here: https://github.com/silverstripe/silverstripe-framework/pull/9064

**Pull request merged**